### ISSUE TITLE
Do not save messages from blocked contact into local database

### ISSUE THREAD

**Issue by csb0730** [deltachat/deltachat-core#230]
In case You use Your mail account in parallel with DC and a normal MUA it's possible that You have messages from a contact which You don't want to see and handle at DC local device.

For that use case You can start a chat with that contact and block it after creating. So You don't see the messages any more. So far ok, but:

in the background DC saves all messages from the blocked contact into the local database!

IMHO I would expect that messages from a blocked contact are not saved locally as long as a contact is blocked.

You can check this behaviour in blocking a known contact, send a message from that afterwards, then unblock the contact and You can see the message immediately.

**Comment by r10s** [comment]
the reason why the message is saved is to avoid downloading it twice - roughly spoken, we need the Message-ID of all messages from the server for this purpose.

we could save the message but not the content and do not show it on unblocking - but this would make things more complicated as needed and maybe not easier to understand for the user. eg. after just blocking a contact, messages are not deleted - if you unblock the contact, messages from this contacts are shown again. i am not sure that deleting _messages_ on blocking _contacts_ is an expected behaviour.

#244 is related

**Comment by csb0730** [comment]
Simply (maybe download it) but throw it away (locally) after that as long as a contact is blocked.

I think that would be the common expectation for a blocked contact !
No need to delete it from server!

Keep things simple!

**Comment by testbird** [comment]
Allowing to track only message IDs where the message itself is not needed or wanted (anymore) and discarded makes sense, to avoid downloads and to allow reduce space usage.

It might even be a requirement to allow syncing with a reduced local storage history (messages remain on server)?

> sort of this may also be implemented to avoid downloading large files or so.

That could be a very good idea! If for example for short message texts that fit completely into the subject line can be detected (maybe by `-[] letting short texts that are complete end with a special character like |`).
Deltachat can then know it was sufficient to just download the listing with the subject lines, and store only the message ID with a property like "text known". And if attachments are sent in separate messages, this could also be the way to avoid having to always download attachments, allowing to only download them when on wifi, or on request.

**Comment by testbird** [comment]
> leave a copy of just the message-id locally for a given time

The clean up could work in the same way as with db entries of complete messages (i.e. a loop removing the oldest entry flagged "synced" (even if it is a "not downloaded" message"), until the space or time requirement is met again https://github.com/deltachat/deltachat-core/issues/164#issuecomment-417868402)

**Comment by r10s** [comment]
superseded by https://github.com/deltachat/deltachat-core/issues/597
